lib/trento_web/controllers/v1/sap_system_controller.ex

```diff
@@ -38,4 +45,81 @@ defmodule TrentoWeb.V1.SapSystemController do
 
     render(conn, "databases.json", databases: databases)
   end
+
+  operation :delete_application_instance,
+    summary: "Delete application instance",
+    description:
+      "Delete the application instance identified by the provided data if it is absent",
+    parameters: [
+      id: [
+        in: :path,
+        required: true,
+        type: %OpenApiSpex.Schema{type: :string, format: :uuid}
+      ],
+      host_id: [
+        in: :path,
+        required: true,
+        type: %OpenApiSpex.Schema{type: :string, format: :uuid}
+      ],
+      instance_number: [
+        in: :path,
+        required: true,
+        type: %OpenApiSpex.Schema{type: :string}
+      ]
+    ],
+    responses: [
+      no_content: "The application instance has been deregistered",
+      not_found: NotFound.response(),
+      unprocessable_entity: UnprocessableEntity.response()
+    ]
+
+  @spec delete_application_instance(Plug.Conn.t(), map) :: {:error, any} | Plug.Conn.t()
+  def delete_application_instance(conn, %{
+        id: sap_system_id,
+        host_id: host_id,
+        instance_number: instance_number
+      }) do
+    case SapSystems.deregister_application_instance(sap_system_id, host_id, instance_number) do
+      :ok -> send_resp(conn, 204, "")
+      {:error, error} -> {:error, error}
```

Comment:
We could go with the next, if this is what you mean, but it wouldn't change much:
```
with :ok <- SapSystems.deregister_application_instance(sap_system_id, host_id, instance_number) do
  send_resp(conn, 204, "")
end
```
Otherwise, you need to pattern match the value, otherwise the `case` raises an error

Reply:
yep, that's what we do in some other controllers.